### PR TITLE
Adding 1-dimensional version of RationalQuadraticSpline and tests for conditional blocks

### PR DESCRIPTION
**Pull request by RussellALA**
- Added new InvertibleModule RationalQuadraticSplines_1D that can be used as invertible learnable transform when couplings are not desirable (e.g. when the feature dimension is 1). 
- Restructured the BinnedSpline class s.t. it works as an InvertibleModule instead of a Coupling
- Added BinnedSplineCoupling class to replace the old BinnedSpline
- Added conditional SequenceINN tests in spline_tests.py

### CODE REVIEW

File: FrEIA/modules/splines/rational_quadratic.py [hunk 1]
```diff
@@ -44,6 +45,65 @@ def _spline2(self, x: torch.Tensor, parameters: Dict[str, torch.Tensor], rev: bo
         return rational_quadratic_spline(x, left, right, bottom, top, deltas_left, deltas_right, rev=rev)
 
 
+class RationalQuadraticSpline_1D(BinnedSpline):
```
Comment (fdraxler) [inline FrEIA/modules/splines/rational_quadratic.py:48]: I think the name might be misleading… 1D suggests that one only can have 1D transformations, but this is not the case

Comment (RussellALA) [reply FrEIA/modules/splines/rational_quadratic.py:48]: Renamed to ElementwiseRationalQuadraticSpline

Comment (fdraxler) [reply FrEIA/modules/splines/rational_quadratic.py:48]: Looks good to me!

**Review by LarsKue — APPROVED** [review]
Code looks good. I like the restructured BinnedSpline. Might be worth looking into refactoring the overall structure of InvertibleModule vs Coupling in general (including the file structure). But I think we should relay that to the future and merge this as-is.

**Pull request merged**